src/pages/cart/Item.tsx

```diff
@@ -0,0 +1,94 @@
+import { useEffect, useState } from "react";
+import { formatPrice } from "../../utils/format";
+import { CartItem } from "../../utils/types";
+import { SlArrowUp, SlArrowDown } from "react-icons/sl";
+import { useQueryClient } from "@tanstack/react-query";
+import { useUpdateCart } from "./useUpdateCart";
+import { useDeleteItem } from "./useDeleteItem";
+
+type ItemProps = {
+  item: CartItem;
+};
+
+function Item({ item }: ItemProps) {
+  const [quantity, setQuantity] = useState(item.quantity);
+  const queryClient = useQueryClient();
+
+  const { error: updateItemError, mutate: updateItem } = useUpdateCart(
+    {
+      ...item,
+      quantity,
+    },
+    {
+      onSuccess: () => {
+        queryClient
+          .invalidateQueries({ queryKey: ["cart"] })
+          .catch(console.error);
+      },
+    }
+  );
+
+  const { error: deleteItemError, mutate: deleteItem } = useDeleteItem({
+    onSuccess: () => {
+      queryClient
+        .invalidateQueries({ queryKey: ["cart"] })
+        .catch(console.error);
+    },
+  });
+
+  useEffect(() => {
+    updateItem();
```

Comment:
Consider passing variables here instead of to the hook, I think it's slightly more idiomatic.